docs/reference/ingest/processors/geo-grid.asciidoc

```diff
@@ -0,0 +1,250 @@
+In order to populate that index using `geotile` and `geohex` fields, we would need to define
+two ingest processors:
+
+[source,console]
+--------------------------------------------------
+PUT geocells
+{
+  "mappings": {
+    "properties": {
+      "geocell": {
+        "type": "geo_shape"
+      }
+    }
+  }
+}
+
+PUT _ingest/pipeline/geotile2shape
+{
+  "description": "translate rectangular z/x/y geotile to bounding box",
+  "processors": [
+    {
+      "geo_grid": {
+        "field": "geocell",
+        "tile_type": "geotile"
+      }
+    }
+  ]
+}
+
+PUT _ingest/pipeline/geohex2shape
+{
+  "description": "translate H3 cell to polygon",
+  "processors": [
+    {
+      "geo_grid": {
+        "field": "geocell",
+        "tile_type": "geohex",
+        "target_format": "wkt"
+      }
+    }
+  ]
+}
+--------------------------------------------------
+
+Using the two pipelines above, we can attempt to index documents into the `geocells` index.
+The `geocell` field will be the string version of either a rectangular tile with format `z/x/y` or an H3 cell address,
+depending on which ingest processor we use when indexing the document.
+The resulting geometry will be represented and indexed as a <<geo-shape,`geo_shape`>> field in either
+http://geojson.org[GeoJSON] or the https://docs.opengeospatial.org/is/12-063r5/12-063r5.html[Well-Known Text] format.
+
+==== Example: Rectangular geo-tile with envelope in GeoJSON
+
+In this example a `geocell` field with a value defined in `z/x/y` format is indexed as a
+http://geojson.org[GeoJSON Envelope] since the ingest-processor above was defined with default `target_format`.
+
+[source,console]
+--------------------------------------------------
+PUT geocells/_doc/1?pipeline=geotile2shape
+{
+  "geocell": "4/8/5"
+}
+
+GET geocells/_doc/1
+--------------------------------------------------
+// TEST[continued]
+
+The response shows how the ingest-processor has replaced the `geocell` field with an indexable `geo_shape`:
+
+[source,console-result]
+--------------------------------------------------
+{
+ "_index": "geocells",
+  "_id": "1",
+  "_version": 1,
+  "_seq_no": 0,
+  "_primary_term": 1,
+  "found": true,
+  "_source": {
+    "geocell": {
+      "type": "Envelope",
+      "coordinates": [
+        [ 0.0, 55.77657301866769 ],
+        [ 22.5, 40.979898069620134 ]
+      ]
+    }
+  }
+}
+--------------------------------------------------
+// TESTRESPONSE[s/"_version": \d+/"_version": $body._version/ s/"_seq_no": \d+/"_seq_no": $body._seq_no/ s/"_primary_term": 1/"_primary_term": $body._primary_term/]
+
+image:images/spatial/geogrid_tile.png[Kibana map with showing the geotile at 4/8/5 and its four child cells]
+
+==== Example: Hexagonal geo-tile with polygon in WKT format
+
+In this example a `geocell` field with an H3 string address is indexed as a
+https://docs.opengeospatial.org/is/12-063r5/12-063r5.html[WKT Polygon], since the ingest-processor explicitly
+defined the `target_format` above.
```

Comment:
```suggestion
https://docs.opengeospatial.org/is/12-063r5/12-063r5.html[WKT Polygon], since this ingest processor explicitly
defined the `target_format`.
```